dnf/cli/cli.py

```diff
@@ -222,6 +222,9 @@ def do_transaction(self, display=()):
         if trans:
             msg = self.output.post_transaction_output(trans)
             logger.info(msg)
+        list_bunch = dnf.cli.output._make_lists(trans, self._goal)
+        if [tsi._active for tsi in list_bunch.failed]:
+            raise dnf.exceptions.Error(_('Transaction failed'))
```

Comment:
Why not `[tsi._active for tsi in trans if tsi.op_type == dnf.transaction.FAIL]`? And is it really necessary to call `._active` ?